### PR TITLE
HPCC-22265 Support FieldFilter syntax field!=[sets] for != matching

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@rpastrana fyi
@richardkchapman please review once smoke test has completed.

**Comment by richardkchapman** [comment]
@richardkchapman

File: rtl/eclrtl/rtlnewkey.cpp [hunk 1]
```diff
@@ -2058,24 +2068,13 @@ IFieldFilter * deserializeFieldFilter(unsigned fieldId, const RtlTypeInfo & type
         {
             char * end;
             size32_t subLength = strtoul(src+1, &end, 10);
+            //This could support !, but this is a legacy syntax : is now specified as part of the set
             if (*end != '=')
                 UNIMPLEMENTED_X("Expected =");
 
-            //Created a truncated type
```
Comment (richardkchapman) [inline rtl/eclrtl/rtlnewkey.cpp]: This refactoring seems unrelated to the feature being implemented? I assume it is using code (subStrinfFieldFilter) that was not available when this was first implemented?

Comment (richardkchapman) [reply rtl/eclrtl/rtlnewkey.cpp]: Somewhat concerned it's using a different type from what it used to

Comment (ghalliday) [reply rtl/eclrtl/rtlnewkey.cpp]: I think the reason was that the test code now calls this function: deserializeFieldFilter, and the way this function handled substrings didn't match the way they were handled in the test code.  I thought the test code was correct, but I will double check again.

Comment (ghalliday) [reply rtl/eclrtl/rtlnewkey.cpp]: Without the change the regression test fails.  I still need to check whether the change is correct though...

Comment (ghalliday) [reply rtl/eclrtl/rtlnewkey.cpp]: I think I would keep this change - this code is used when string filters are processed in dafilesrv - which means it should not assume it has been serialized by a fieldvalue.  (Without the change some of the test generate an internal error.)  It is also in a deprecated syntax, so unlikely to cause any issues.

**Comment by ghalliday** [comment]
@richardkchapman would take a is.
